packages/core/src/kernelTemplate.js

```diff
@@ -1,5 +1,18 @@
 // LavaMoat Prelude
 (function () {
+  // identify the globalRef
+  const globalRef = (typeof globalThis !== 'undefined') ? globalThis : (typeof self !== 'undefined') ? self : (typeof global !== 'undefined') ? global : undefined
+  if (!globalRef) {
+    throw new Error('Lavamoat - unable to identify globalRef')
+  }
+
+  // polyfill globalThis
```

Comment:
why is this necessary?

Reply:
don't know, that part isn't new though, i just moved it from a few lines later to here because I needed the determined `globalRef` object to get SNOW from it, and since that happenes in the outer scope I had to move it upwards a little bit.

Reply:
`globalThis` was not common early in lavamoat's history. it may be prevalent enough to remove

Reply:
afba9f65d

Reply:
> [afba9f6](https://github.com/LavaMoat/LavaMoat/commit/afba9f65d3b4e7404b5b374198c10838ef841458)

Seems like this belongs as a separate change (unless there's a link I'm missing)?

I propose soft-deprecating it for now and fully removing it in a subsequent release.

https://github.com/LavaMoat/LavaMoat/pull/459